File: ejercicios/src/main/java/co/edu/sena/ejercicios/ejercicio11/partes/Torre.java
package co.edu.sena.ejercicios.ejercicio11.partes;

import co.edu.sena.ejercicios.ejercicio11.Parte;

public class Torre extends Parte {

    private DiscoDuro discoDuro;
    private MotherBoard tarjetaMadre;
    private RAM tarjetaMemoria;
    private Fuente fuente;
    private Procesador procesador;
    private Disipador disipador;
    private LectorCD lectorCD;
    private LectorDVD lectorDVD;
    private LectorBR lectorBR;
    private LectorDisquette lectorDisquette;
    private TarjetaSonido tarjetaSonido;
    private TarjetaVideo tarjetaVideo;
    private Bateria bateria;

    public Torre(String serial, String marca, DiscoDuro discoDuro, MotherBoard tarjetaMadre, RAM tarjetaMemoria, Fuente fuente, Procesador procesador, Disipador disipador) {
        super(serial, marca);
        this.discoDuro = discoDuro;
        this.tarjetaMadre = tarjetaMadre;
        this.tarjetaMemoria = tarjetaMemoria;
        this.fuente = fuente;
        this.procesador = procesador;
        this.disipador = disipador;
    }

    public DiscoDuro getDiscoDuro() {
        return discoDuro;
    }

    public void setDiscoDuro(DiscoDuro discoDuro) {
        this.discoDuro = discoDuro;
    }

    public MotherBoard getTarjetaMadre() {
        return tarjetaMadre;
    }

    public void setTarjetaMadre(MotherBoard tarjetaMadre) {
        this.tarjetaMadre = tarjetaMadre;
    }

    public RAM getTarjetaMemoria() {
        return tarjetaMemoria;
    }

    public void setTarjetaMemoria(RAM tarjetaMemoria) {
        this.tarjetaMemoria = tarjetaMemoria;
    }

    public Fuente getFuente() {
        return fuente;
    }

    public void setFuente(Fuente fuente) {
        this.fuente = fuente;
    }

    public Procesador getProcesador() {
        return procesador;
    }

    public void setProcesador(Procesador procesador) {
        this.procesador = procesador;
    }

    public Disipador getDisipador() {
        return disipador;
    }

    public void setDisipador(Disipador disipador) {
        this.disipador = disipador;
    }

    public LectorCD getLectorCD() {
        return lectorCD;
    }

    public void setLectorCD(LectorCD lectorCD) {
        this.lectorCD = lectorCD;
    }

    public LectorDVD getLectorDVD() {
        return lectorDVD;
    }

    public void setLectorDVD(LectorDVD lectorDVD) {
        this.lectorDVD = lectorDVD;
    }

    public LectorBR getLectorBR() {
        return lectorBR;
    }

    public void setLectorBR(LectorBR lectorBR) {
        this.lectorBR = lectorBR;
    }

    public LectorDisquette getLectorDisquette() {
        return lectorDisquette;
    }

    public void setLectorDisquette(LectorDisquette lectorDisquette) {
        this.lectorDisquette = lectorDisquette;
    }

    public TarjetaSonido getTarjetaSonido() {
        return tarjetaSonido;
    }

    public void setTarjetaSonido(TarjetaSonido tarjetaSonido) {
        this.tarjetaSonido = tarjetaSonido;
    }

    public TarjetaVideo getTarjetaVideo() {
        return tarjetaVideo;
    }

    public void setTarjetaVideo(TarjetaVideo tarjetaVideo) {
        this.tarjetaVideo = tarjetaVideo;
    }

    public Bateria getBateria() {
        return bateria;
    }

    public void setBateria(Bateria bateria) {
        this.bateria = bateria;
    }
}

File: typescript/src/ejercicio11/partes/torre.ts
import {Parte} from "../parte";
import {MotherBoard} from "./mother.board";
import {Ram} from "./ram";
import {Fuente} from "./fuente";
import {Procesador} from "./procesador";
import {Disipador} from "./disipador";
import {LectorCd} from "./lector.cd";
import {LectorDvd} from "./lector.dvd";
import {LectorBR} from "./lector.b.r";
import {LectorDisquette} from "./lector.disquette";
import {TarjetaSonido} from "./tarjeta.sonido";
import {TarjetaVideo} from "./tarjeta.video";
import {Bateria} from "./bateria";
import {DiscoDuro} from "./disco.duro";

export class Torre extends Parte {
    private _discoDuro: DiscoDuro;
    private _tarjetaMadre: MotherBoard;
    private _tarjetaMemoria: Ram;
    private _fuente: Fuente;
    private _procesador: Procesador;
    private _disipador: Disipador;
    private _lectorCD: LectorCd;
    private _lectorDVD: LectorDvd;
    private _lectorBR: LectorBR;
    private _lectorDisquette: LectorDisquette;
    private _tarjetaSonido: TarjetaSonido;
    private _tarjetaVideo: TarjetaVideo;
    private _bateria: Bateria;

    constructor(serial: string, marca: string, discoDuro: DiscoDuro, tarjetaMadre: MotherBoard, tarjetaMemoria: Ram, fuente: Fuente, procesador: Procesador, disipador: Disipador) {
        super(serial, marca);
        this._discoDuro = discoDuro;
        this._tarjetaMadre = tarjetaMadre;
        this._tarjetaMemoria = tarjetaMemoria;
        this._fuente = fuente;
        this._procesador = procesador;
        this._disipador = disipador;
    }


    get discoDuro(): DiscoDuro {
        return this._discoDuro;
    }

    set discoDuro(value: DiscoDuro) {
        this._discoDuro = value;
    }

    get tarjetaMadre(): MotherBoard {
        return this._tarjetaMadre;
    }

    set tarjetaMadre(value: MotherBoard) {
        this._tarjetaMadre = value;
    }

    get tarjetaMemoria(): Ram {
        return this._tarjetaMemoria;
    }

    set tarjetaMemoria(value: Ram) {
        this._tarjetaMemoria = value;
    }

    get fuente(): Fuente {
        return this._fuente;
    }

    set fuente(value: Fuente) {
        this._fuente = value;
    }

    get procesador(): Procesador {
        return this._procesador;
    }

    set procesador(value: Procesador) {
        this._procesador = value;
    }

    get disipador(): Disipador {
        return this._disipador;
    }

    set disipador(value: Disipador) {
        this._disipador = value;
    }

    get lectorCD(): LectorCd {
        return this._lectorCD;
    }

    set lectorCD(value: LectorCd) {
        this._lectorCD = value;
    }

    get lectorDVD(): LectorDvd {
        return this._lectorDVD;
    }

    set lectorDVD(value: LectorDvd) {
        this._lectorDVD = value;
    }

    get lectorBR(): LectorBR {
        return this._lectorBR;
    }

    set lectorBR(value: LectorBR) {
        this._lectorBR = value;
    }

    get lectorDisquette(): LectorDisquette {
        return this._lectorDisquette;
    }

    set lectorDisquette(value: LectorDisquette) {
        this._lectorDisquette = value;
    }

    get tarjetaSonido(): TarjetaSonido {
        return this._tarjetaSonido;
    }

    set tarjetaSonido(value: TarjetaSonido) {
        this._tarjetaSonido = value;
    }

    get tarjetaVideo(): TarjetaVideo {
        return this._tarjetaVideo;
    }

    set tarjetaVideo(value: TarjetaVideo) {
        this._tarjetaVideo = value;
    }

    get bateria(): Bateria {
        return this._bateria;
    }

    set bateria(value: Bateria) {
        this._bateria = value;
    }


    toString(): string {
        return "";
    }
}

